File: src/main/java/com/tiobe/plugins/intellij/actions/CancelAnalysis.java
package com.tiobe.plugins.intellij.actions;

import com.intellij.openapi.actionSystem.AnActionEvent;
import com.tiobe.plugins.intellij.analyzer.TICSAnalyzer;

/**
 * {@link com.intellij.openapi.actionSystem.AnAction} that cancels the currently running TICS analysis.
 */
public class CancelAnalysis extends com.intellij.openapi.actionSystem.AnAction {
    @Override
    public void actionPerformed(final AnActionEvent anActionEvent) {
        TICSAnalyzer.getInstance().stop();
    }

    @Override
    public void update(final AnActionEvent event) {
        event.getPresentation().setEnabled(TICSAnalyzer.getInstance().isRunning() && !TICSAnalyzer.getInstance().isStopping());
    }
}

File: src/main/kotlin/com/tiobe/plugins/intellij/actions/CancelAnalysis.kt
package com.tiobe.plugins.intellij.actions

import com.intellij.openapi.actionSystem.ActionUpdateThread
import com.intellij.openapi.actionSystem.AnAction
import com.intellij.openapi.actionSystem.AnActionEvent
import com.tiobe.plugins.intellij.analysis.ProcessRunner
import com.tiobe.plugins.intellij.analysis.ProcessState

class CancelAnalysis : AnAction() {
    override fun actionPerformed(e: AnActionEvent) {
        ProcessRunner.stop()
    }

    override fun update(e: AnActionEvent) {
        e.presentation.isVisible =
            ProcessState.isRunning() && !ProcessState.isStopping()
    }

    override fun getActionUpdateThread(): ActionUpdateThread {
        return ActionUpdateThread.BGT
    }
}